src/vuexOffline.js

```diff
@@ -105,124 +105,145 @@ export default class {
+
+          ~index && state.list.splice(index, 1)
         }
       },
 
       // actions
       actions: {
-        create: async ({ commit }, payload = {}) => {
+        create: async ({ commit }, { payload }) => {
           try {
-            const response = await this.pouchDBService.save(resource, payload)
-            const result = { ...response, ...payload }
+            const uuid = new Uuid()
+            const document = await collection.insert({ uuid: uuid.create(), ...payload })
+            const parsedDocument = document.toJSON()
 
             commit('setErrors', { model: 'onCreate' })
-            commit('setItemList')
+            commit('setItemList', parsedDocument)
 
-            return Promise.resolve(result)
+            return parsedDocument
           } catch (error) {
             commit('setErrors', { model: 'onCreate', hasError: true })
             return Promise.reject(error)
           }
         },
 
         replace: async ({ commit }, { payload, id } = {}) => {
-          try {
-            const response = (await this.pouchDBService.find(resource, id || payload.id))[resource][0]
-
-            if (!response) {
-              throw new FormatError({
-                status: {
-                  code: '404',
-                  text: 'not found!'
-                }
-              })
-            }
-
-            const mergedResponse = merge(response, payload)
-            await this.pouchDBService.save(resource, mergedResponse)
+          return save({ commit }, { payload, id, model: 'onReplace' })
+        },
 
-            commit('setErrors', { model: 'onReplace' })
-            commit('replaceItem', mergedResponse)
-            return Promise.resolve(mergedResponse)
-          } catch (error) {
-            commit('setErrors', { model: 'onReplace', hasError: true })
-            return Promise.reject(error)
-          }
+        update: async ({ commit }, { payload, id } = {}) => {
+          return save({ commit }, { payload, id, model: 'onUpdate' })
         },
 
         fetchSingle: async ({ commit }, { form, id, params, url } = {}) => {
-          try {
-            const result = (await this.pouchDBService.find(resource, id))[resource][0]
-
-            if (!result) {
-              throw new FormatError({
-                status: {
-                  code: '404',
-                  text: 'Not found!'
-                }
-              })
+          if (!id && form) {
+            return {
+              data: {
+                status: { code: 200 },
+                fields: fieldsWithRelationOptions
+              }
             }
+          }
 
-            commit('replaceItem', result)
-            commit('setErrors', { model: 'onFetchSingle' })
+          try {
+            const document = await collection.findOne(id).exec()
+
+            if (!document) {
+              throw new FormatError({ status: { code: '404', text: 'Not found!' } })
```

Comment:
Vamos padronizar o `text` como `Not found`, sem exclamação e com o `N` maiúsculo, atualiza nos outros lugares.

Reply:
feito